File: DSA/Searching/binary_search.cpp
#include <bits/stdc++.h>
using namespace std;

int search(vector<int>& nums, int target) {
        int l=0,r=nums.size()-1;
        return bs(nums,target,l,r);
    }
    int bs(vector<int> v,int x,int l,int r){
        int mid=0;
        if(l<=r){
            mid=(r+l)/2;
        
        if(v[mid]==x){
            return mid;
        }
        if(v[mid]<x){
            return bs(v,x,mid+1,r);
        }
        return bs(v,x,l,mid-1);
        }
        return -1;
    }

int main()
{
	int arr[] = { 2, 3, 4, 10, 40 };
	int x = 10;
	int n = sizeof(arr) / sizeof(arr[0]);
	int result = search(arr, 0, n - 1, x);
	(result == -1) ? cout << "Element is not present in array"
				: cout << "Element is present at index " << result;
	return 0;
}

File: Python/binary_search/binary_search.py

def binary_search(arr:list, low:int, high:int, x:int) -> list:

    if high >= low:
        mid = (high + low) // 2
        if arr[mid] == x:
            return mid
        elif arr[mid] > x:
            return binary_search(arr, low, mid - 1, x)
        else:
            return binary_search(arr, mid + 1, high, x)
    else:
        return -1
 
arr:list = []
for _ in range(int(input("Enter the max index: "))):
  arr.append(input(f"Enter {_} element: "))

x = input("Enter element to search: ")
 

result = binary_search(arr, 0, len(arr)-1, x)
 
if result != -1:
    print("Element is present at index", str(result))
else:
    print("Element is not present in array")
